test/unit/spec/PasswordExpired_spec.js

```diff
@@ -114,24 +116,71 @@ function (Q, _, $, OktaAuth, LoginUtil, Util, PasswordExpiredForm, Beacon, Expec
         });
       });
       itp('calls processCreds function before saving a model', function () {
-        return setup().then(function (test) {
+        var processCredsSpy = jasmine.createSpy('processCredsSpy');
+        return setup({ processCreds: processCredsSpy })
+        .then(function (test) {
           $.ajax.calls.reset();
-          processCredsSpy.calls.reset();
           test.setNextResponse(resSuccess);
           submitNewPass(test, 'oldpwd', 'newpwd', 'newpwd');
           expect(processCredsSpy.calls.count()).toBe(1);
           expect(processCredsSpy).toHaveBeenCalledWith({
             username: 'inca@clouditude.net',
             password: 'newpwd'
           });
+          return Expect.waitForSpyCall(test.successSpy);
+        })
+        .then(function() {
+          expect($.ajax.calls.count()).toBe(1);
```

Comment:
In this case I'd move the expect processCredsSpy logic to this block (i.e. after we're done waiting for success to be called) - it works because Q.Promise calls its function synchronously, but better to be explicit.

i.e.
```javascript
var processCredsSpy = jasmine.createSpy('processCredsSpy');
return setup({ processCreds: processCredsSpy })
.then(function (test) {
  $.ajax.calls.reset();
  test.setNextResponse(resSuccess);
  submitNewPass(test, 'oldpwd', 'newpwd', 'newpwd');
  return Expect.waitForSpyCall(test.successSpy);
})
.then(function() {
  expect(processCredsSpy.calls.count()).toBe(1);
  expect(processCredsSpy).toHaveBeenCalledWith({
    username: 'inca@clouditude.net',
    password: 'newpwd'
  });
  expect($.ajax.calls.count()).toBe(1);
});
```